tests/test-leak-client.lua

```diff
@@ -85,12 +85,23 @@ local steps = {
             -- Test that we have a client and that it's invalid (tostring()
             -- causes an "invalid object" error)
             local success, msg = pcall(function() tostring(objs[1]) end)
-            assert(not success)
+            assert(not success, msg)
             assert(msg:find("invalid object"), msg)
 
-            -- Check that it is garbage-collectable
-            collectgarbage("collect")
-            assert(#objs == 0)
+            -- On GitHub Actions, it can take a while for clients to be killed
+            -- properly.
```

Comment:
This makes no sense to me. With "clients to be killed properly" you mean that you are waiting for something to close its X11 window? If so, then a `while` loop in Lua will not work for waiting for that. The C code does not handle X11 events until you return to it, thus this code cannot wait until some window closed.

The whole test suite stuff exists to "wait for something". So, if you want to wait for something, then add another step to this.

Also, this whole code is inside a `if #client.get() == 0`, so at this point we are sure that there are no open clients.

So, what is this even waiting for? The only thing that happens in this loop is that you repeatedly run the garbage collector and run `sleep 1` in between. The `sleep 1` will only cause delays and have no other effects besides that. So, that leaves `collectgarbage("collect")`. Did something change so that we now need more iterations of the GC before something becomes collectable? What keeps clients alive now?

:-/